source/fundamentals/connection.txt

```diff
@@ -0,0 +1,636 @@
+``MongoClients.create()``. You can specify one or more compression
+algorithms, separating multiple algorithms with commas:
+
+.. code-block:: java
+
+   ConnectionString connectionString = new ConnectionString("mongodb://localhost/?compressors=snappy,zlib,zstd");
+   MongoClient mongoClient = MongoClients.create(connectionString);
+
+Specify compression algorithms using the following strings:
+
+- "snappy" for `Snappy <https://google.github.io/snappy/>`__ compression.
+
+- "zlib" for `Zlib <https://zlib.net/>`__ compression.
+
+- "zstd" for `Zstandard <https://github.com/facebook/zstd/>`__ compression.
+
+.. _compression-mongoclientsettings:
+
+Enable Compression via MongoClientSettings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable compression with :java-sync-api:`MongoClientSettings
+<com/mongodb/MongoClientSettings.html>`, pass the
+``compressorList()`` builder method a list of ``MongoCompressor``
+instances. You can specify one or more compression algorithms in the list:
+
+.. code-block:: java
+   :emphasize-lines: 2-4
+
+   MongoClientSettings settings = MongoClientSettings.builder()
+        .compressorList(Arrays.asList(MongoCompressor.createSnappyCompressor(),
+                                      MongoCompressor.createZlibCompressor(),
+                                      MongoCompressor.createZstdCompressor()))
+        .build();
+   MongoClient client = MongoClients.create(settings);
+
+.. _compression-dependencies:
+
+Compression Algorithm Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The JDK supports `Zlib <https://zlib.net/>`__ compression natively, but
+`Snappy <https://google.github.io/snappy/>`__ and
+`Zstandard <https://github.com/facebook/zstd/>`__ depend upon open source
+implementations. See 
+`snappy-java <https://github.com/xerial/snappy-java>`__ and
+`zstd-java <https://github.com/luben/zstd-jni>`__ for details.
+
+.. _connect-replica-set:
+
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration of instances provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname and port numbers
+of one or more members of the replica set separated by commas.
+
+.. code-block:: none
+
+   mongodb://host1:27017,host2:27017,host3:27017/?replicaSet=myRs
+
+.. note::
+
+   Although it is possible to connect to a replica set deployment by
+   providing only a single host, you should provide the driver with the
+   full list to ensure that it is able to connect even if one host fails.
+
+.. _connection-options:
+
+Connection Options
+------------------
+
+This section explains MongoDB connection and authentication options
+supported by the Java driver. You can pass the connection options as
+parameters of the connection URI to specify the behavior of the client.
+
+.. list-table::
```

Comment:
Nitpick: You can probably leave out mention of "Java" when describing the driver as that is implied on this page and explicitly stated in this section.